Resources/config/services.yml

```diff
@@ -29,3 +31,12 @@ services:
             - '@event_dispatcher'
         tags:
             - { name: pim_datagrid.extension.mass_action.handler, alias: custom_entity_mass_delete }
+
+    pim_custom_entity.repository.attribute.factory:
+        class: '%pim_custom_entity.repository.attribute.factory.class%'
+        arguments:
+            - '@doctrine.orm.default_entity_manager'
+
+    pim_custom_entity.repository.attribute:
+        class: '%pim_custom_entity.repository.attribute.class%'
+        factory: ['@pim_custom_entity.repository.attribute.factory', 'getCustomAttributesRepository']
```

Comment:
Please create a single attribute repository that extends EntityRepository. It takes as parameter EntityManager + attribute class name and calls parent construct with attribute repository + em->getEntityClassMetadata($className)
It is done like in Akeneo ;)